.github/workflows/python.yml

```diff
@@ -56,41 +56,41 @@ jobs:
       matrix:
         name:
           - conda-python-docs
-          - conda-python-3.9-nopandas
-          - conda-python-3.8-pandas-1.0
-          - conda-python-3.10-pandas-latest
-          - conda-python-3.10-no-numpy
+          - conda-python-3.10-nopandas
+          - conda-python-3.9-pandas-1.5
+          - conda-python-3.11-pandas-latest
+          - conda-python-3.11-no-numpy
         include:
           - name: conda-python-docs
-            cache: conda-python-3.9
+            cache: conda-python-3.10
             image: conda-python-docs
-            title: AMD64 Conda Python 3.9 Sphinx & Numpydoc
-            python: 3.9
-          - name: conda-python-3.9-nopandas
-            cache: conda-python-3.9
+            title: AMD64 Conda Python 3.10 Sphinx & Numpydoc
+            python: "3.10"
+          - name: conda-python-3.10-nopandas
+            cache: conda-python-3.10
             image: conda-python
-            title: AMD64 Conda Python 3.9 Without Pandas
-            python: 3.9
-          - name: conda-python-3.8-pandas-1.0
-            cache: conda-python-3.8
+            title: AMD64 Conda Python 3.10 Without Pandas
+            python: "3.10"
+          - name: conda-python-3.9-pandas-1.5
+            cache: conda-python-3.9
             image: conda-python-pandas
-            title: AMD64 Conda Python 3.8 Pandas 1.0
-            python: 3.8
-            pandas: "1.0"
-            numpy: 1.16
-          - name: conda-python-3.10-pandas-latest
-            cache: conda-python-3.10
+            title: AMD64 Conda Python 3.9 Pandas 1.5
+            python: 3.9
+            pandas: "1.5"
+            numpy: 1.24
```

Comment:
Given this is meant to test somewhat the minimum supported version (I assume), I would maybe take 1.19 for numpy, if we look at what numpy version supported Python 3.9. 
Of course, if we look at something similar as pandas 1.5 in terms of when it was released, then 1.23 or 1.24 also makes sense.

We also declare `"numpy >= 1.16.6"` in pyproject.toml, which can be updated.

(given we don't really have issues regularly with older numpy, I think I personally lean to be more conservative)

Reply:
FWIW if we want to keep this PR conservative regarding other deps, and just update numpy / pandas to the minimal required to allow bumping Python, then it would be pandas 1.1 and numpy 1.19. 
And then we can have a separate issue / PR about properly bumping pandas and numpy (and cleaning up the code for it), because I do support bumping pandas to 1.5.